ports/stm32/rfcore.c

```diff
@@ -762,4 +761,40 @@ STATIC mp_obj_t rfcore_sys_hci(size_t n_args, const mp_obj_t *args) {
 }
 MP_DEFINE_CONST_FUN_OBJ_VAR_BETWEEN(rfcore_sys_hci_obj, 3, 4, rfcore_sys_hci);
 
+STATIC void rfcore_ble_hci_response_to_buffer(void *env, const uint8_t *buf, size_t len) {
+    // mp_buffer_info_t *respbufinfo = (mp_buffer_info_t *)env;
+    DEBUG_printf("rfcore_ble_hci_response_to_buffer len 0x%x\n", len);
+    mp_obj_t *rsp = (mp_obj_t *)env;
+    *rsp = mp_obj_new_bytes(buf, len);
+}
+
+STATIC mp_obj_t rfcore_ble_hci(size_t n_args, const mp_obj_t *args) {
+    if (ipcc_mem_dev_info_tab.fus.table_state == MAGIC_IPCC_MEM_INCORRECT) {
+        mp_raise_OSError(MP_EINVAL);
+    }
+    mp_buffer_info_t bufinfo = {0};
+    // Can accept either just a raw buffer to send, or (ogf, ocf, buffer, <timeout>) to assemble and send.
+    if (mp_get_buffer(args[0], &bufinfo, MP_BUFFER_READ)) {
+        // Poll for completion rather than wait for IRQ->scheduler.
+        LL_C1_IPCC_DisableReceiveChannel(IPCC, IPCC_CH_BLE);
+
+        rfcore_ble_hci_cmd(bufinfo.len, bufinfo.buf);
+
+        mp_obj_t rsp = mp_const_none;
+        parse_hci_info_t parse = { rfcore_ble_hci_response_to_buffer, &rsp, false };
+        tl_ble_wait_resp(&parse);
+        return rsp;
+    } else {
+        mp_int_t ogf = mp_obj_get_int(args[0]);
+        mp_int_t ocf = mp_obj_get_int(args[1]);
+        mp_get_buffer_raise(args[2], &bufinfo, MP_BUFFER_READ);
+        size_t len = tl_ble_hci_cmd_resp(HCI_OPCODE(ogf, ocf), bufinfo.buf, bufinfo.len);
```

Comment:
this should be `ssize_t`

Reply:
This whole section is now removed as it was just to support the other (unused) function branch.